jenkins/Jenkinsfile

```diff
@@ -65,147 +88,15 @@ pipeline {
-    // that purpose, see the following stage, "UI Tests"
-    stage('UI BDD Tests') {
-      steps {
-        sh './gradlew runBehaveTests'
-        sh './gradlew generateCucumberReport'
-      }
-      post {
-        always {
-          junit 'build/test-results/bdd_ui/*.xml'
-        }
-      }
-    }
-
-    // This set of tests investigates the functionality of the UI.
-    // Note that this is separate fom the UI BDD Tests, which
-    // only focuses on essential capability and therefore only
-    // covers a small subset of the possibilities of UI behavior.
-    stage('UI Tests') {
-        steps {
-            sh 'cd src/ui_tests/java && ./gradlew clean test'
-        }
-        post {
-            always {
-                junit 'src/ui_tests/java/build/test-results/test/*.xml'
-            }
-        }
-    }
-
-    // Run OWASP's "DependencyCheck". https://owasp.org/www-project-dependency-check/
-    // You are what you eat - and so it is with software.  This
-    // software consists of a number of software by other authors.
-    // For example, for this project we use language tools by Apache,
-    // password complexity analysis, and several others.  Each one of
-    // these might have security bugs - and if they have a security
-    // bug, so do we!
-    //
-    // DependencyCheck looks at the list of known
-    // security vulnerabilities from the United States National Institute of
-    // Standards and Technology (NIST), and checks if the software
-    // we are importing has any major known vulnerabilities. If so,
-    // the build will halt at this point.
-    stage('Security: Dependency Analysis') {
-      steps {
-         sh './gradlew dependencyCheckAnalyze'
-      }
-    }
-
-    // Run Jmeter performance testing https://jmeter.apache.org/
-    // This test simulates 50 users concurrently using our software
-    // for a set of common tasks.
-    stage('Performance Tests') {
-      steps {
-         sh './gradlew runPerfTests'
-      }
-    }
-
-    // Runs mutation testing against some subset of our software
-    // as a spot test.  Mutation testing is where bugs are seeded
-    // into the software and the tests are run, and we see which
-    // tests fail and which pass, as a result.
-    //
-    // what *should* happen is that where code or tests are altered,
-    // the test should fail, shouldn't it? However, it sometimes
-    // happens that no matter how code is changed, the tests
-    // continue to pass, which implies that the test wasn't really
-    // providing any value for those lines.
-    stage('Mutation Tests') {
-      steps {
-         sh './gradlew pitest'
+        environment {
+          // Override JAVA_HOME to use JDK 11 for this stage
+          JAVA_HOME = '/usr/lib/jvm/java-11-openjdk'
+          PATH = "${JAVA_HOME}/bin:${PATH}"
+        }      
+      
+      steps{
+        sh './gradlew sonarqube -Dsonar.host.url=http://sonarqube:9000 -Dsonar.login="admin" -Dsonar.password="ensf400"'
```

Comment:
The SonarQube password can be changed based on what you set initially. Alternatively, you may manage this password using Jenkins Credentials and do a token replacement here.